src/main/java/com/jagrosh/discordipc/entities/Callback.java

```diff
@@ -27,7 +27,7 @@
  */
 public class Callback
 {
-    private final Runnable success;
+    private final Consumer<Packet> success;
```

Comment:
I'd be interested to know the use cases behind providing the Packet as the success consumer?

Reply:
It is useful for the extended (non-rich-presence) API, for instance for requesting a user's info and reacting to the response.

Though it seems that oauth scope is not made available yet.